src/plugins/ui_stateful/clipboard.ts

```diff
@@ -156,10 +165,13 @@ export class ClipboardPlugin extends UIPlugin {
         break;
       }
       case "PASTE_FROM_OS_CLIPBOARD":
-        this.state = new ClipboardOsState(cmd.text, this.getters, this.dispatch, this.selection);
-        this.state.paste(cmd.target);
-        this.lastPasteState = this.state;
-        this.status = "invisible";
+        if (this.allowPaste) {
```

Comment:
`PASTE_FROM_OS_CLIPBOARD` should ignore `this.allowPaste`, since it won't have the correct value. Imagine a situation when we cup/pasted cometing, `this.allowPaste` is false; but then we copy something in another application and try to paste it in the spreadsheet. We don't want to block the paste.

In practice, `PASTE` will be used by grid when the clipboard content was copied from the spreadhseet, `PASTE_FROM_OS_CLIPBOARD` will be used when the clipboard content comes from somewhere else.